common/service/config/ringpop.go

```diff
@@ -125,13 +129,21 @@ func newRingpopFactory(rpConfig *Ringpop) (*RingpopFactory, error) {
 }
 
 // CreateRingpop is the implementation for RingpopFactory.CreateRingpop
-func (factory *RingpopFactory) CreateRingpop(ch *tchannel.Channel) (*ringpop.Ringpop, error) {
+func (factory *RingpopFactory) CreateRingpop(dispatcher *yarpc.Dispatcher) (*ringpop.Ringpop, error) {
 
 	discoveryProvider, err := newDiscoveryProvider(factory.config)
 	if err != nil {
 		return nil, err
 	}
 
+	t := dispatcher.Inbounds()[0].Transports()[0].(*tchannel.ChannelTransport)
```

Comment:
move this into a separate method that takes dispatcher as input and returns channel as output